lib/rucio/daemons/abacus/rse.py

```diff
@@ -62,32 +64,38 @@ def rse_update(once=False):
         try:
             # Heartbeat
             heartbeat = live(executable=executable, hostname=hostname, pid=pid, thread=current_thread)
+            prepend_str = 'abacus-rse[%i/%i] ' % (heartbeat['assign_thread'], heartbeat['nr_threads'])
+            logger = formatted_logger(logging.log, prepend_str + '%s')
+            logger(logging.INFO, 'Abacus-rse started')
 
             # Select a bunch of rses for to update for this worker
             start = time.time()  # NOQA
             rse_ids = get_updated_rse_counters(total_workers=heartbeat['nr_threads'],
                                                worker_number=heartbeat['assign_thread'])
-            logging.debug('Index query time %f size=%d' % (time.time() - start, len(rse_ids)))
+            logger(logging.DEBUG, 'Index query time %f size=%d' % (time.time() - start, len(rse_ids)))
 
             # If the list is empty, sent the worker to sleep
             if not rse_ids and not once:
-                logging.info('rse_update[%s/%s] did not get any work' % (heartbeat['assign_thread'], heartbeat['nr_threads'] - 1))
+                logger(logging.INFO, 'did not get any work')
                 time.sleep(10)
             else:
                 for rse_id in rse_ids:
                     if graceful_stop.is_set():
                         break
                     start_time = time.time()
                     update_rse_counter(rse_id=rse_id)
-                    logging.debug('rse_update[%s/%s]: update of rse "%s" took %f' % (heartbeat['assign_thread'], heartbeat['nr_threads'] - 1, rse_id, time.time() - start_time))
+                    logger(logging.DEBUG, 'counter update of rse "%s" took %f' % (rse_id, time.time() - start_time))
+                    start_time = time.time()
+                    update_replica_counts(rse_id=rse_id)
```

Comment:
I think this wouldn't work here.

The problem is that the `get_updated_rse_counters` gives you RSEs which have new replicas added or removed. But it doesn't give you RSEs whose replicas changed (from no tombstone to obsolete tombstone for example) - So it might be if nothing else happens with an RSE, that the obsolete/expired counter would never be updated.

I think this needs to be done completely separately and I would probably just do it in one big query, and not go RSE by RSE.